timeout.go

```diff
@@ -61,25 +59,25 @@ func NewTimerWheel(min, max time.Duration) *TimerWheel {
 	// timeout
 	wLen := int((max / min) + 2)
 
-	tw := TimerWheel{
+	tw := TimerWheel[T]{
 		wheelLen:      wLen,
-		wheel:         make([]*TimeoutList, wLen),
+		wheel:         make([]*TimeoutList[T], wLen),
 		tickDuration:  min,
 		wheelDuration: max,
-		expired:       &TimeoutList{},
+		expired:       &TimeoutList[T]{},
 	}
 
 	for i := range tw.wheel {
-		tw.wheel[i] = &TimeoutList{}
+		tw.wheel[i] = &TimeoutList[T]{}
 	}
 
 	return &tw
 }
 
-// Add will add a firewall.Packet to the wheel in it's proper timeout
-func (tw *TimerWheel) Add(v firewall.Packet, timeout time.Duration) *TimeoutItem {
+// Add will add an item to the wheel in its proper timeout
+func (tw *TimerWheel[T]) Add(v T, timeout time.Duration) *TimeoutItem[T] {
 	// Check and see if we should progress the tick
-	tw.advance(time.Now())
+	tw.Advance(time.Now())
```

Comment:
Lost to the sands of time, timeout_system does not advance during Add. It has the line of code to advance commented out. Git has no history why. (I can speculate that Add is called in the hot path, and wheel work is desired to be avoided in the hot path.)
It is a difference between the timeout.go timer wheel and timeout_system timer wheel which might be a wrinkle in using one for both - unless the advance during Add becomes some kind of config parameter.

Reply:
Updated to allow the `HandshakeManager` and `ConnectionManager` to continue with the previous behavior of `Advance`ing on their own